src/engine/routing_algorithms/map_matching.cpp

```diff
@@ -199,9 +199,9 @@ SubMatchingList mapMatching(SearchEngineData<Algorithm> &engine_working_data,
 
             const auto haversine_distance = util::coordinate_calculation::haversineDistance(
                 prev_coordinate, current_coordinate);
-            // assumes minumum of 4 m/s
+            // assumes minumum of 5 km/h == 1.39 m/s
             const EdgeWeight weight_upper_bound =
-                ((haversine_distance + max_distance_delta) / 4.) * facade.GetWeightMultiplier();
+                (max_network_distance / 1.39) * facade.GetWeightMultiplier();
```

Comment:
Here can be a potential problem if two timestamps are equal and `haversine_distance > 0`, so `step_time`,  `max_network_distance` and `weight_upper_bound` will be 0 and `getNetworkDistance` will fail to find a path, because `weight` must be strictly less than `weight_upper_bound`.

At the moment coordinates in traces with `haversine_distance=0` and `step_time=0` lead to a gap in trace.